src/meta/src/hummock/vacuum.rs

```diff
@@ -112,6 +113,15 @@ impl VacuumManager {
         let mut batch_idx = 0;
         let batch_size = 500usize;
         let mut sent_batch = Vec::with_capacity(objects_to_delete.len());
+        if self.env.opts.enable_hummock_data_archive {
+            if !objects_to_delete.is_empty() {
+                warn!(
+                    ?objects_to_delete,
+                    "objects not deleted in vacuum due to enable_hummock_data_archive"
+                );
+            }
+            return Ok(objects_to_delete);
```

Comment:
Though the returned value is not used, returning the full `objects_to_delete` while they are not deleted seems strange to me.